File: rb/features.rb
# NovayaGazeta SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module NovayaGazetaFeatures
  def self.make_feature(name)
    case name
    when "base"
      NovayaGazetaBaseFeature.new
    when "ratelimit"
      NovayaGazetaRatelimitFeature.new
    when "retry"
      NovayaGazetaRetryFeature.new
    when "test"
      NovayaGazetaTestFeature.new
    when "timeout"
      NovayaGazetaTimeoutFeature.new
    else
      NovayaGazetaBaseFeature.new
    end
  end
end

File: py/features.py
# NovayaGazeta SDK feature factory

from feature.base_feature import NovayaGazetaBaseFeature
from feature.test_feature import NovayaGazetaTestFeature


def _make_feature(name):
    features = {
        "base": lambda: NovayaGazetaBaseFeature(),
        "test": lambda: NovayaGazetaTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
